spec/controllers/admin/posts_controller_spec.rb

```diff
@@ -0,0 +1,100 @@
+
+require 'rails_helper'
+
+describe Admin::PostsController do
+
+  before (:each) do
+    @user = create :user, :admin
+    sign_in @user
+  end
+
+  let(:valid_attributes) { attributes_for(:post) }
+
+  describe 'GET' do
+
+    let(:post) { create :post }
+
+    describe 'GET index' do
+      it 'assigns all admin_posts as @admin_posts' do
+        get :index
+        expect(assigns(:posts)).to eq([post])
+      end
+    end
+
+    describe 'GET show' do
+      it 'assigns the requested admin_post as @admin_post' do
+        get :show, id: post.to_param
+        expect(assigns(:post)).to eq(post)
+      end
+    end
+
+    describe 'GET new' do
+      it 'assigns a new admin_post as @admin_post' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.